vendor_consignment_stock/model/warehouse.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+#    Author: Leonardo Pistone
+#    Copyright 2014 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from openerp import models, api, fields, exceptions
+from openerp.tools.translate import _
+
+
+class Warehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    buy_vci_to_resupply = fields.Boolean(
+        'Purchase from VCI to resupply this warehouse',
+        help="This warehouse can contain Vendor-Supplied-Inventory (VCI) that "
+        "have to be bought before being sold.", default=True)
+    buy_vci_pull_id = fields.Many2one('procurement.rule', 'BUY VCI rule')
+
+    @api.model
+    def _get_buy_vci_pull_rule(self, warehouse):
+        route_model = self.env['stock.location.route']
+        try:
+            buy_vci_route = self.env.ref(
+                'vendor_consignment_stock.route_warehouse0_buy_vci')
+        except:
+            buy_vci_route = route_model.search([
+                ('name', 'like', _('Buy VCI'))
+            ])
+        if not buy_vci_route:
+            raise exceptions.Warning(_(
+                'Can\'t find any generic Buy VCI route.'))
+
+        return {
+            'name': self._format_routename(warehouse, _(' Buy VCI')),
```

Comment:
You can remove that space before `Buy VCI` it is already in `_format_routename`

    def _format_routename(self, cr, uid, obj, name, context=None):
        return obj.name + ': ' + name

By the way don't you want to extend `get_routes_dict` to use key value?